packages/@ember/-internals/metal/lib/property_events.ts

```diff
@@ -24,12 +24,18 @@ let deferred = 0;
   @for @ember/object
   @param {Object} obj The object with the property that will change
   @param {String} keyName The property key (or path) that will change.
-  @param {Meta} meta The objects meta.
+  @param {Meta} [_meta] The objects meta.
+  @param {Boolean} [_shouldFirePropertyDidChange=true] Whether to fire `PROPERTY_DID_CHANGE`
   @return {void}
   @since 3.1.0
   @public
 */
-function notifyPropertyChange(obj: object, keyName: string, _meta?: Meta | null): void {
+function notifyPropertyChange(
+  obj: object,
+  keyName: string,
+  _meta?: Meta | null,
+  _shouldFirePropertyDidChange = true
```

Comment:
Discussed this with @rwjblue this morning, and we realized a better solution here would be to pass the value optionally as the last argument to `notifyPropertyChange` here. This way, `PROPERTY_DID_CHANGE` can receive the value for this use case, and otherwise lookup the value if its undefined, and we don't have to call `PROPERTY_DID_CHANGE` again later on and copy all that logic.

Reply:
Thanks for the review! I just updated the PR to pass the `value` through to `notifyPropertyChange`, then to `PROPERTY_DID_CHANGE` to avoid calling `get`.